src/renderers/dom/client/eventPlugins/SimpleEventPlugin.js

```diff
@@ -483,6 +483,10 @@ for (var type in topLevelEventsToDispatchConfig) {
 var ON_CLICK_KEY = keyOf({onClick: null});
 var onClickListeners = {};
 
+function getDictionaryKey(inst) {
+  return '.' + inst._rootNodeID;
+}
+
```

Comment:
This is still a "numeric" key as considered by V8. (I tried that and it doesn't help.)